Update the community ecosystem page

adamghill opened the pull request:
Update the community ecosystem page with an eye toward:

- cleaning up some of the copy
- splitting up the resource section to have sub-headings for better organization
- adding a table of contents or quick links to go to a particular section since the page is long
- making it clear that sub-headings can be linked to
- adding visual interest to the page

Based on this original analysis of the page: https://docs.google.com/document/d/1h3IAMafx5K30DdRQYx8OiiSwbpiqQlJC3uP2oP6et9Q/edit?tab=t.0#heading=h.la6v9fpeqmn.

LilyFirefly commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -6,38 +6,58 @@
     <li>
       <a href="https://awesomedjango.org">Awesome Django</a>
-      &mdash; A curated list of awesome things related to Django.
+      &mdash; A curated list of awesome resouces for Django.
```
```suggestion
      &mdash; A curated list of awesome resources for Django.
```

tim-schilling commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -6,38 +6,58 @@
-      &mdash; Django Girls is a non-profit organization and a community that empowers and helps women to organize free, one-day programming workshops by providing tools, resources and support.
+      &mdash; Django Girls is a non-profit organization and a community that empowers and helps women to organize free,
+      one-day programming workshops by providing tools, resources and support.
     </li>
```
Seems like Djangonaut Space should belong in this category as well.

adamghill commented:
<img width="1241" height="920" alt="image" src="https://github.com/user-attachments/assets/0e938e08-3b16-4b49-bb17-550766e91c58" />

I don't love the right sidebar styles with a list of links here. Previously it was a `dt`/`dd`, but adding a description for every link seems a little overkill. And just a `dt` adds a border to visually separate them. The `li` isn't big enough, though. I'll keep tweaking this unless someone has a strong opinion about it.

tim-schilling commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -66,147 +87,278 @@ <h2 id="resources">Resources</h2>
+    </li>
   </ul>
 
-  <h2>More Information</h2>
```
Did you have an idea of what to do with this @adamghill?

LilyFirefly replied:
No objection from me - I don't mind adding a `<details>` section either.

tim-schilling reviewed (commented):
Thank you Adam for iterating on this!

adamghill commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -66,147 +87,278 @@ <h2 id="resources">Resources</h2>
 
-  <p>We could never list everything but here are some packages that the Steering Council all agree are
-    incredibly useful and well supported. We use many of these in all of our projects.</p>
+  <h2 id="packages">Packages</h2>
```
`Community Packages` makes sense to me. WIll update.

tim-schilling commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -4,40 +4,60 @@
-      &mdash; Django Girls is a non-profit organization and a community that empowers and helps women to organize free, one-day programming workshops by providing tools, resources and support.
+      &mdash; Django Girls is a non-profit organization and a community that empowers and helps women to organize free,
+      one-day programming workshops by providing tools, resources and support.
     </li>
```
```suggestion
    </li>
    <li>
      <a href="https://djangonaut.space">Djangonaut Space</a>
      &mdash; Djangonaut Space is a community that helps individuals learn how to contribute to Django and other open-source packages. It hosts multiple, free, 8-week group mentoring programs each year. It's meant for people who wish to level up their current Django code contributions and potentially take on leadership roles in Django in the future.
    </li>
```

tim-schilling commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -6,38 +6,58 @@
-    <!-- Name, Short Description, Link — Roughly this order       -->
+  <!-- Name, Short Description, Link — roughly this order -->
+
+  <h3 id="third-party-packages">Third-party Packages</h3>
```
What about Collections? Those two sites are comparable to this page really.

adamghill commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -66,147 +87,278 @@ <h2 id="resources">Resources</h2>
-  <hr />
-  <h3 id="exploratory-packages">Exploratory packages needing feedback</h3>
-  <p>The Steering Council is exploring the viability of these packages and would benefit from your help. Please try them out and open an issue with your feedback!</p>
+  <h3 id="exploratory-packages">Exploratory packages</h3>
```
I can add this in! What would you like the description to be?

tim-schilling reviewed (approved):
I'm good with the changes to the ecosystem.html page, pending the resolution of the wording on the async cursors (there's a suggestion which is probably good enough).

This shouldn't be merged without eyeballs from the website WG since the change impacts more than the just the ecosystem page.

sarahboyce commented on djangoproject/templates/base_community.html:
```diff
@@ -25,42 +25,30 @@
     <h2 class="visuallyhidden" id="aside-header">{% translate "Additional Information" %}</h2>
     {% donation_snippet %}
 
-    <h3>{% translate "More Help" %}</h3>
```
I think reading the PR, I feel like perhaps you want to add a `{% block contents %}{% endblock %}` which is overwritten on the ecosystem page to add navigation similar to what would exist on a docs page (see https://github.com/django/djangoproject.com/blob/8d230390ceb35c892cc9110968412f42f8811f3c/docs/templates/docs/doc.html#L145)

sarahboyce reviewed (commented):
Thank you! I added some comments on the changes to the complimentary section 👍

adamghill commented on djangoproject/templates/base_community.html:
```diff
@@ -25,42 +25,30 @@
+        <a href="https://stackoverflow.com/questions/tagged/django" rel="nofollow">StackOverflow</a>
+      </li>
+      <li>
+        <a href="irc://irc.libera.chat/django" rel="nofollow">#django IRC Channel</a>
```
Makes sense to me. Can you give me a link for the official Discord? I'm in there, but can't figure out a link. 😵

sarahboyce commented on djangoproject/templates/base_community.html:
```diff
@@ -25,42 +25,30 @@
-      <dt><a href="irc://irc.libera.chat/django">{% translate "#django IRC Channel" %}</a></dt>
-      <dd>{% translate "Chat with other Django users like it's 1999" %}</dd>
-    </dl>
+    <h3>{% translate "Social Media" %}</h3>
```
OK I see 👍
Yes, I think removing the IRC makes sense also. No reason for it to be so prominent 

I don't really know what's best for all community pages or for the ecosystem page specifically. So, personally I would be tempted to leave it alone for this PR and pull out any changes into a different PR so the discussions on this content doesn't impede merging in the improvements to the main section of the ecosystem page

adamghill replied:
Works for me! I reverted my changes to `base_community.html` in 58a1c89f. I did leave the IRC link in there for now just to not muddy up the waters.

adamghill commented:
> stick with the "¶" character that's used in the documentation for consistency
> keeping consistency with the docs

Yeah, ¶ is definitely the better approach. For now, I manually added links for each header in 8889d011. In the future JavaScript could do this automatically so it'd less error-prone, but that could be a separate PR. I think adding a transition opacity on hover to the ¶ would also be a nice touch, but that'd be a website-wide thing so I'm also going to skip that for now. 😄

adamghill commented:
I think this PR has addressed all of the feedback so far. Let me know if there is anything else I can look at or additional changes to make. Thanks!

Edit: one thing I just realized is that the original page had a list of links to social media profiles which I removed from this version since I was hoping to add them to the sidebar. I can 1) add them back in to the ecosystem page, 2) create a new PR to add them to the sidebar, or 3) something else entirely.

sarahboyce commented on djangoproject/templates/aggregator/ecosystem.html:
```diff
@@ -3,41 +3,78 @@
 
 {% block content %}
 
+  <h1>Community Ecosystem</h1>
```
```suggestion
  <h1>Package Ecosystem</h1>
```
Should we align the page title to how it is described/linked on the index page:
https://github.com/django/djangoproject.com/blob/8d230390ceb35c892cc9110968412f42f8811f3c/djangoproject/templates/aggregator/index.html#L35

adamghill replied:
I can add an `id` to this `h1` and an `a` with a `headerlink` class here if you want me to?

sarahboyce replied:
Oh good thought 🤔 
Yes, I suppose that's similar to the docs and makes sense here also

adamghill replied:
Done in cd596ff. ✔️

sarahboyce reviewed (approved):
Thank you! Added one thought around page navigation but happy with the changes :+1:

adamghill commented:
Actually, one more question: would you like me to rebase/squash my commits to make this PR cleaner?

sarahboyce commented:
> Actually, one more question: would you like me to rebase/squash my commits to make this PR cleaner?

This is my personal preference but I think it can be squashed and merged :+1:

adamghill commented:
> This is my personal preference but I think it can be squashed and merged 👍

Ok, I rebased it for a cleaner history. However, I cannot merge it -- maybe waiting on approval from @django/django-website?

SaptakS reviewed (commented):
This PR looks great to me! Just one small nitpick related to the anchor links added for headings. I am usually not a fan of `title` attribute since the tooltip created by it is not keyboard accessible. I would probably prefer just having an aria-label instead. Also I like the format of the link labels to be "Link to section titled Community Ecosystem". Or something like `aria-label="Link to this heading" aria-describedby="community-ecosystem"`. Both of this will help provide context of the link to the screen reader users.
